tasks/tests/test_create_task.py

```diff
@@ -0,0 +1,154 @@
+                                       role=Role.EMPLOYEE,
+                                       team_id=other_team)
+        employee.save()
+        return employee
+
+    """
+    Test add new task
+    """
+    def test_add_to_db(self, manager, employee):
+        Task.create_task(title="TestTask",
+                         assignee=employee,
+                         assigner=manager,
+                         priority=Priority.CRITICAL,
+                         status=Status.BACKLOG,
+                         description="This is a test task")
+        assert len(Task.objects.all()) == 1
+
+    """
+    Test that a user who is not manager cannot assign task to other
+    teams employees.
+    """
+    def test_assigned_by_non_manager(self, manager, employee):
+        assert employee.role != Role.MANAGER
+        with pytest.raises(ValueError):
+            Task.create_task(title="TestTask",
+                             assignee=manager,
+                             assigner=employee,
+                             priority=Priority.CRITICAL,
+                             status=Status.BACKLOG,
+                             description="This is a test task")
+        assert len(Task.objects.all()) == 0
+
+    """
+    Test that title is required when creating a task.
+    """
+    def test_no_title(self, manager, employee):
+        with pytest.raises(ValueError):
+            Task.create_task(title="",
+                             assignee=employee,
+                             assigner=manager,
+                             priority=Priority.CRITICAL,
+                             status=Status.BACKLOG,
+                             description="This is a test task")
+        assert len(Task.objects.all()) == 0
+
+    """
+    Test that priority must be a valid enum value
+    """
+    def test_invalid_priority(self, manager, employee):
+        with pytest.raises(ValueError):
+            Task.create_task(title="TestTask",
+                             assignee=employee,
+                             assigner=manager,
+                             priority='INVALID',
+                             status=Status.BACKLOG,
+                             description="This is a test task")
+        assert len(Task.objects.all()) == 0
+
+    """
+    Test that status must be a valid enum value
+    """
+    def test_invalid_status(self, manager, employee):
+        with pytest.raises(ValueError):
+            Task.create_task(title="TestTask",
+                             assignee=employee,
+                             assigner=manager,
+                             priority=Priority.CRITICAL,
+                             status='INVALID',
+                             description="This is a test task")
+        assert len(Task.objects.all()) == 0
+
+    """
+    Test that manager cannot assign task to other teams employees
+    """
+    def test_assign_other_team(self, manager, employee_other_team):
```

Comment:
maybe instead of a repeat of most of the parameters in each negative test, you can consider using parametrize
https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
I'll look into it

Reply:
It seems that it's impossible to use fixtures as parameters of parametrize. 
For the parametrize will be helpful for me in this case I need to parametrize the assignee and assigner in some of the tests, which as I mentioned seems to be impossible.
Also, found an open issue with this idea on pytestrepo:
https://github.com/pytest-dev/pytest/issues/349#issuecomment-457355792

Reply:
Is there another way I am missing here?

Reply:
I understand the point of using parameterized tests, but this time I can't get it. The example you wrote doesn't test what it should.
The difference between:
"test_add_task_to_db"
"test_assigned_by_non_manager"
is that the assignee and created_by attributes in the user creation are swapped, which are fixtures.
The problem I mentioned is that to do the parameterized tests with these tests, I have to use the fixtures as changing parameters. This kind of parameterization seems to be impossible with pytest.mark.parametrized as I mentioned in one of the comments  above.

Reply:
I figure it out, sending you what we need to do @orzionpour

Reply:
Resolved.